### PR TITLE
Adds job to provide testing

### PR DESCRIPTION
**Pull request by Cwagne17**
Job implemented npm run test. This can be expanded to support more testing suites as we grow our testing framework

### CODE REVIEW

**Review by StarChaser44 — APPROVED** [review]

**Review by JosueMolinaMorales — APPROVED** [review]

File: src/jobs/backend-tests.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+parameters:
+  # Not Required Parameters
+  docker-image:
+    type: string
+    default: "cimg/base:stable"
+    description: |
+      The docker image that will be used as an executor
+      if isNode is true, node/default is recommended
+  isNode:
```
Comment (pzalep1) [inline src/jobs/backend-tests.yml:12]: When we get to Python tests will we add an isPython parameter?

**Review by EVogel1999 — APPROVED** [review]

File: src/jobs/backend-tests.yml (hunk 1 above)
Comment (Cwagne17) [reply src/jobs/backend-tests.yml:12]: Right, we can just expand it like that then just add the when condition and what is nessecary.

File: src/jobs/test-app.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        Node: <<parameters.isNode>>"
+  - when:
+      condition: <<parameters.isNode>>
+      steps:
+        - node/install
+        - node/install-packages
+        - run:
+            name: Run Tests
+            command: >
+              npm run test
```
Comment (EVogel1999) [inline src/jobs/test-app.yml:39]: Do you think we could add another job that takes in the command to test?  The ```npm run test``` here being the default if none are present.  There's only a few repos that do this.  I.e. ```npm run test:lighthouse```

**Review by EVogel1999 — COMMENTED** [review]
Just one comment to make this a bit more extendable

**Review by pzalep1 — APPROVED** [review]

**Review by AStoerkel1 — APPROVED** [review]
looks good
